sdk/src/main/scala/com/horizen/SidechainState.scala

```diff
@@ -11,50 +11,59 @@ import com.horizen.certnative.BackwardTransfer
 import com.horizen.utils.{BlockFeeInfo, ByteArrayWrapper, BytesUtils, FeePaymentsUtils, MerkleTree, TimeToEpochUtils, WithdrawalEpochInfo, WithdrawalEpochUtils}
-import sparkz.crypto.hash.Blake2b256
-import sparkz.util.{ModifierId, SparkzLogging, bytesToId}
+import com.horizen.validation.crosschain.{CrossChainValidatorContainer, CrossChainValidatorKey}
+import com.horizen.validation.crosschain.receiver.{CrossChainRedeemMessageBodyToValidate, CrossChainRedeemMessageValidator}
+import com.horizen.validation.crosschain.sender.{CrossChainMessageBodyToValidate, CrossChainMessageValidator}
 import sparkz.core._
 import sparkz.core.transaction.state._
+import sparkz.crypto.hash.Blake2b256
+import sparkz.util.{ModifierId, SparkzLogging, bytesToId}
+
 import java.io.File
 import java.math.{BigDecimal, MathContext}
 import java.util
 import java.util.{ArrayList => JArrayList, HashMap => JHashMap, Optional => JOptional}
-
-import com.horizen.sc2sc.{CrossChainMessage, CrossChainMessageHash, CrossChainMessageImpl, Sc2ScConfigurator}
-
 import scala.collection.JavaConverters._
 import scala.collection.mutable.ListBuffer
 import scala.util.{Failure, Success, Try}
 
 
-class SidechainState private[horizen] (stateStorage: SidechainStateStorage,
-                                       forgerBoxStorage: SidechainStateForgerBoxStorage,
-                                       utxoMerkleTreeProvider: SidechainStateUtxoMerkleTreeProvider,
-                                       val params: NetworkParams,
-                                       val sc2scConfig : Sc2ScConfigurator,
-                                       override val version: VersionTag,
-                                       val applicationState: ApplicationState)
+class SidechainState private[horizen](stateStorage: SidechainStateStorage,
+                                      forgerBoxStorage: SidechainStateForgerBoxStorage,
+                                      utxoMerkleTreeProvider: SidechainStateUtxoMerkleTreeProvider,
+                                      val params: NetworkParams,
+                                      val sc2scConfig: Sc2ScConfigurator,
+                                      sidechainSettings: SidechainSettings,
+                                      override val version: VersionTag,
+                                      val applicationState: ApplicationState)
   extends AbstractState[SidechainTypes#SCBT, SidechainBlockHeader, SidechainBlock, SidechainState]
     with TransactionValidation[SidechainTypes#SCBT]
     with ModifierValidation[SidechainBlock]
     with SidechainTypes
     with NodeState
     with SparkzLogging
     with UtxoMerkleTreeView
-    with NetworkParamsUtils
-{
+    with NetworkParamsUtils {
 
+  protected val crossChainValidatorContainer: CrossChainValidatorContainer = new CrossChainValidatorContainer(
```

Comment:
i would keep it simpler: we are defining a key-value structure but anyone wil be able to add more entries from outside the state
We can instead use a similar structure as AbstractSidechainNodeViewHolder.semanticBlockValidators :  

- define a trait Sc2ScBlockValidator similar to com.horizen.validation.SemanticBlockValidator with a method validate(block: PMOD)
-  here initialize a Seq of the above trait, with two implementations: one for output crosschainmessages and one for input

Reply:
That's a good point, gonna fix it